sdc/tests/test_series.py

```diff
@@ -4216,52 +4223,164 @@ def test_impl(S):
+        hpat_func2 = self.jit(test_impl_literal)
+
+        S = pd.Series(['ac', 'c', 'cb', 'ca', None, 'da', 'cc', 'ddd', 'd'])
+        for ascending in (False, True):
+            with self.subTest(literal_value='no', ascending=ascending):
+                pd.testing.assert_series_equal(hpat_func1(S, ascending), test_impl(S, ascending))
+
+        with self.subTest(literal_value='yes'):
+            pd.testing.assert_series_equal(hpat_func2(S), test_impl_literal(S))
+
+    @skip_sdc_jit('Old-style impl returns array but not Series')
+    def test_series_sort_values_invalid_axis(self):
+        """Verifies Series.sort_values method raises with invalid value of parameter 'axis'"""
+        def test_impl(S, param_value):
+            return S.sort_values(axis=param_value)
         hpat_func = self.jit(test_impl)
 
-        S = pd.Series(['a', 'd', 'r', 'cc'])
-        pd.testing.assert_series_equal(test_impl(S), hpat_func(S))
+        S = pd.Series(['ac', 'c', 'cb', 'ca', None, 'da', 'cc', 'ddd', 'd'])
+        unsupported_values = [1, 'columns', 'abcde']
+        for axis in unsupported_values:
+            with self.assertRaises(Exception) as context:
+                test_impl(S, axis)
+            pandas_exception = context.exception
 
-    def test_series_sort_values_index1(self):
-        def test_impl(A, B):
-            S = pd.Series(A, B)
-            return S.sort_values()
+            self.assertRaises(type(pandas_exception), hpat_func, S, axis)
+
+    @skip_sdc_jit('Old-style impl returns array but not Series')
+    @skip_numba_jit('TODO: inplace sorting is not implemented yet')
+    def test_series_sort_values_inplace(self):
+        def test_impl(S):
+            S.sort_values(inplace=True)
+            return S
+        hpat_func = self.jit(test_impl)
+
+        S1 = pd.Series(['ac', 'c', 'cb', 'ca', None, 'da', 'cc', 'ddd', 'd'])
+        S2 = S1.copy()
+        pd.testing.assert_series_equal(hpat_func(S1), test_impl(S2))
+
+    @skip_sdc_jit('Old-style impl returns array but not Series')
+    def test_series_sort_values_kind(self):
+        """Verifies Series.sort_values method support of parameter 'kind'
+           on a unnamed Series of different dtypes and default index"""
+        def test_impl(A, param_value):
+            # FIXME: use literally(kind) because, numpy.argsort is supported by Numba with literal kind value only
+            # and literally when applied inside sort_values overload_method impl is not working due to some bug in Numba
+            return A.sort_values(kind=literally(param_value))
+        hpat_func = self.jit(test_impl)
+
+        # using sequences of unique values because default sorting algorithm is not stable
+        n = 11
+        data_to_test = [
+            [1, -0., 0.2, -3.7, np.inf, np.nan, -1.0, 2/3, 21.2, -np.inf, 9.99],
+            np.arange(-10, 20, 1),
+            np.unique(np.random.ranf(n)),
+            np.unique(np.random.randint(0, 100, n)),
```

Comment:
Let's set up seed before generating randoms here and in below test.

Reply:
@densmirn Thank you, corrected.